spec/wordlist_repo_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'spec_helper'
+require 'ronin/wordlists/wordlist_repo'
+
+describe Ronin::Wordlists::WordlistRepo do
+  subject { described_class.new(path) }
+
+  let(:path) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures', 'wordlists')) }
+
+  describe "#initialize" do
+    it "must initialize #path and #name" do
+      expect(subject.path).to eq(path)
+      expect(subject.name).to eq("wordlists")
+    end
+  end
+
+  describe "#download" do
+    let(:repo_path) { File.join(Dir.pwd, "") }
+    let(:url)       { "https://www.example.com" }
+
+    context "when git clone succeed" do
+      it "must return new wordlist repo" do
+        expect(described_class).to receive(:system).with('git', 'clone', '--depth', '1', '--', url, repo_path).and_return(true)
+
+        expect(described_class.download(url)).to be_kind_of(Ronin::Wordlists::WordlistRepo)
+      end
+    end
+
+    context "when git clone failed" do
+      it "must raise DownloadFailed error" do
+        expect(described_class).to receive(:system).with('git', 'clone', '--depth', '1', '--', url, repo_path).and_return(false)
+
+        expect {
+          described_class.download(url)
+        }.to raise_error(Ronin::Wordlists::DownloadFailed, "git command failed: git clone --depth 1 -- #{url} #{repo_path}")
+      end
+    end
+
+    context "when git is not installed" do
+      it "must raise DownloadFailed error" do
+        expect(described_class).to receive(:system).with('git', 'clone', '--depth', '1', '--', url, repo_path).and_return(nil)
+
+        expect {
+          described_class.download(url)
+        }.to raise_error(Ronin::Wordlists::DownloadFailed, "git is not installed on the system")
+      end
+    end
+  end
+
+  describe "#type" do
+    it "must return :git" do
+      expect(subject.type).to eq(:git)
+    end
+  end
+
+  describe "#git?" do
+    context "if wordlist repository use git" do
+      it "must return true" do
+        allow(File).to receive(:directory?).and_return(true)
+
+        expect(subject.git?).to eq(true)
+      end
+    end
+
+    context "if wordlist reposiftory does not use" do
+      it "must return false" do
+        expect(subject.git?).to eq(false)
+      end
+    end
+  end
+
+  describe "#filename" do
+    it "must return the name of the wordlist repo directory" do
+      expect(subject.filename).to eq("wordlists")
+    end
+  end
+
+  describe "#url" do
+    context "if wordlist repository uses git" do
+      let(:url) { Dir.chdir(subject.path) { `git config --get remote.origin.url`.chomp } }
```

Comment:
I try to avoid nesting `{ }`s for readability, and would recommend using `let(:url) do` so the outer block is a `do`/`end` block.